src/test/rgw/rgw_multi/tests.py

```diff
@@ -1314,3 +1315,734 @@ def test_bucket_creation_time():
+    buckets.append(bucketA)
+    bucketB = create_zone_bucket(zcA)
+    buckets.append(bucketB)
+
+    zonegroup_meta_checkpoint(zonegroup)
+
+    # configure sync policy & pipe for only bucketA
+    create_sync_policy_group(c1, "sync-group")
+    create_sync_group_flow_symmetrical(c1, "sync-group", "sync-flow1", zones)
+    args = "--source-bucket " + bucketA.name + " --dest-bucket " + bucketA.name
+
+    create_sync_group_pipe(c1, "sync-group", "sync-pipe", zones, zones, None, args)
+    set_sync_policy_group_status(c1, "sync-group", "enabled")
+    get_sync_policy(c1)
+    zonegroup.period.update(zoneA, commit=True)
+    
+    sync_info(c1)
+
+    # create objects in bucketA & bucketB
+    create_objects(zcA, bucketA, objnames, content)
+    create_object(zcA, bucketB, objnames, content)
+
+    zonegroup_meta_checkpoint(zonegroup)
+    zone_data_checkpoint(zoneB, zoneA)
+
+    # verify if bucketA objects are synced
+    bucket = get_bucket(zcB, bucketA.name)
+    check_objects_exist(bucket, objnames, content)
+
+    # bucketB objects should not be synced
+    bucket = get_bucket(zcB, bucketB.name)
+    check_objects_not_exist(bucket, objnames)
+
+    return
+
+def test_sync_pipe_different_buckets():
+    """
+    test_sync_pipe_different_buckets:
+        directional flow
+        pipe: sync zoneA bucketA to zoneB bucketB
+    """
+
+    zonegroup = realm.master_zonegroup()
+    zonegroup_meta_checkpoint(zonegroup)
+
+    zonegroup_conns = ZonegroupConns(zonegroup)
+
+    (zoneA, zoneB) = zonegroup.zones[0:2]
+    (zcA, zcB) = zonegroup_conns.zones[0:2]
+
+    c1 = zoneA.cluster
+
+    c1.admin(['sync policy get'])
+
+    objnames = [ 'obj1', 'obj2' ]
+    content = 'asdasd'
+    buckets = []
+
+    # create bucketA & bucketB in zoneA
+    bucketA = create_zone_bucket(zcA)
+    buckets.append(bucketA)
+    bucketB = create_zone_bucket(zcA)
+    buckets.append(bucketB)
+
+    zonegroup_meta_checkpoint(zonegroup)
+
+    # configure pipe from zoneA bucketA to zoneB bucketB
+    create_sync_policy_group(c1, "sync-group")
+    create_sync_group_flow_directional(c1, "sync-group", "sync-flow", zoneA.name, zoneB.name)
+    args = '--source-bucket=' + bucketA.name + ' --dest-bucket=' + bucketB.name
+    create_sync_group_pipe(c1, "sync-group", "sync-pipe", zoneA.name, zoneB.name, None, args)
+    set_sync_policy_group_status(c1, "sync-group", "enabled")
+    zonegroup.period.update(zoneA, commit=True)
+    get_sync_policy(c1)
+
+    # create objects in bucketA
+    create_objects(zcA, bucketA, objnames, content)
+
+    zonegroup_meta_checkpoint(zonegroup)
+    zone_data_checkpoint(zoneB, zoneA)
+
+    # verify that objects are synced to bucketB in zoneB
+    # but not to bucketA
+    bucket = get_bucket(zcB, bucketA.name)
+    check_objects_not_exist(bucket, objnames)
+
+    bucket = get_bucket(zcB, bucketB.name)
+    check_objects_exist(bucket, objnames, content)
+
+    return
+
+def test_sync_pipe_multiple_buckets_to_single():
```

Comment:
@yehudasa  ..I tested this (and the below test case - test_sync_pipe_single_bucket_to_multiple) configuration manually. It doesn't seem to be working. The data is not synced. Could you please take a look and let me know if I missed out anything.

Reply:
@soumyakoduri sure, I will take a look

Reply:
@soumyakoduri looking at it now, you should be creating different pipes to different source buckets. See here: https://github.com/yehudasa/ceph/commit/40a751163409da9529e032c3c0995ac9da035820

Also, it's better if you set a wildcard 'allowed' policy at the zonegroup level, and create the bucket sync policy on the bucket. So at the zonegroup level just specify '*' as source and dest buckets, and for this test create two separate pipes on the (preferably target) bucket policy itself. Bucket policy changes don't require period update, only ones that are on the zonegroup.

Reply:
@soumyakoduri also see here for the second part: https://github.com/yehudasa/ceph/commit/d993fa093b705075e96939a52bbab06b5077a595

Reply:
> @soumyakoduri looking at it now, you should be creating different pipes to different source buckets. See here: [yehudasa@40a7511](https://github.com/yehudasa/ceph/commit/40a751163409da9529e032c3c0995ac9da035820)
> 
> Also, it's better if you set a wildcard 'allowed' policy at the zonegroup level, and create the bucket sync policy on the bucket. So at the zonegroup level just specify '*' as source and dest buckets, and for this test create two separate pipes on the (preferably target) bucket policy itself. Bucket policy changes don't require period update, only ones that are on the zonegroup.

Thanks. That worked. So that means instead of creating comma separated bucket list, there should be different pipe created for each <source,dest> bucket combination.. right?

For the second part, I will add them as separate test-cases. Most of the test cases added here are to test functionality at zonegroup level, except for the last one. Will now create corresponding  bucket-level cases for each of them.